nsfw/nsfw.php

```diff
@@ -34,126 +34,125 @@ function nsfw_extract_photos($body)
+	$s .= '</span>';
+	$s .= '<div id="settings_nsfw_expanded" class="settings-block" style="display: none;">';
+	$s .= '<span class="fakelink" onclick="openClose(\'settings_nsfw_expanded\'); openClose(\'settings_nsfw_inflated\');">';
+	$s .= '<h3>' . L10n::t('Not Safe For Work (General Purpose Content Filter)') . '</h3>';
+	$s .= '</span>';
+
+	$s .= '<div id="nsfw-wrapper">';
+	$s .= '<p>' . L10n::t('This addon looks in posts for the words/text you specify below, and collapses any content containing those keywords so it is not displayed at inappropriate times, such as sexual innuendo that may be improper in a work setting. It is polite and recommended to tag any content containing nudity with #NSFW.  This filter can also match any other word/text you specify, and can thereby be used as a general purpose content filter.') . '</p>';
+	$s .= '<label id="nsfw-enable-label" for="nsfw-enable">' . L10n::t('Enable Content filter') . ' </label>';
+	$s .= '<input id="nsfw-enable" type="checkbox" name="nsfw-enable" value="1"' . $enable_checked . ' />';
 	$s .= '<div class="clear"></div>';
-    $s .= '<label id="nsfw-label" for="nsfw-words">' . L10n::t('Comma separated list of keywords to hide') . ' </label>';
-    $s .= '<textarea id="nsfw-words" type="text" name="nsfw-words">' . $words .'</textarea>';
-    $s .= '</div><div class="clear"></div>';
+	$s .= '<label id="nsfw-label" for="nsfw-words">' . L10n::t('Comma separated list of keywords to hide') . ' </label>';
+	$s .= '<textarea id="nsfw-words" type="text" name="nsfw-words">' . $words .'</textarea>';
+	$s .= '</div><div class="clear"></div>';
 
-    $s .= '<div class="settings-submit-wrapper" ><input type="submit" id="nsfw-submit" name="nsfw-submit" class="settings-submit" value="' . L10n::t('Save Settings') . '" /></div>';
+	$s .= '<div class="settings-submit-wrapper" ><input type="submit" id="nsfw-submit" name="nsfw-submit" class="settings-submit" value="' . L10n::t('Save Settings') . '" /></div>';
 	$s .= '<div class="nsfw-desc">' . L10n::t('Use /expression/ to provide regular expressions') . '</div></div>';
-
 	return;
-
 }
 
-function nsfw_addon_settings_post(&$a,&$b) {
-
-	if(! local_user())
+function nsfw_addon_settings_post(&$a, &$b)
+{
+	if (!local_user()) {
 		return;
+	}
 
-	if($_POST['nsfw-submit']) {
-		PConfig::set(local_user(),'nsfw','words',trim($_POST['nsfw-words']));
-		$enable = ((x($_POST,'nsfw-enable')) ? intval($_POST['nsfw-enable']) : 0);
+	if ($_POST['nsfw-submit']) {
+		PConfig::set(local_user(), 'nsfw', 'words', trim($_POST['nsfw-words']));
+		$enable = (x($_POST,'nsfw-enable') ? intval($_POST['nsfw-enable']) : 0);
 		$disable = 1-$enable;
-		PConfig::set(local_user(),'nsfw','disable', $disable);
+		PConfig::set(local_user(), 'nsfw', 'disable', $disable);
 		info(L10n::t('NSFW Settings saved.') . EOL);
 	}
 }
 
-function nsfw_prepare_body(&$a,&$b) {
-
+function nsfw_prepare_body(&$a, &$b)
+{
+	// Don't do the check when there is a content warning
```

Comment:
I think this should be optional. Say someone uses the content filter to filter out trigger words. The posting will be collapsed. Unless the trigger word is used in the description for the content warning, which may be something different as different people react differently to words and their usage.

Reply:
I'm thinking about using the content warning for the "nsfw" functionality. So the addon could simply fill the new field. This would avoid double collapsed posts.

Reply:
For clarification - you want to make the core depend on the RegEx used in the settings of an optional addon to fill the DB?

Reply:
Another thing came to my mind. If one changes the addon settings, all postings would need to be res-canned for the changed terms the content filter should react on-right?

Reply:
> you want to make the core depend on the RegEx used in the settings of an optional addon to fill the DB?

No, the content warning item field would still be populated, but the final display would be driven by the addon, not the Core.

> all postings would need to be res-canned for the changed terms the content filter should react on-right?

The NSFW plugin checks for the specified words on display, there's no timeline-wide scan. Meaning if you never see an item, then its NSFW status will not be re-checked every time the setting is changed.

Reply:
Currently the CW stuff is done very basic. In the future we could improve the design by passing the values to the user interface. And then it would make sense that the "nsfw" addon simply would fill some variable that is otherwise used for CW. Then we had one central functionality to create this hiding and unhiding of stuff.

Reply:
I'm not sure about this approach. The `content-warning` item field is remotely set, while the NSFW and the language filter are locally set, so I don't think both should be merged into a single property. On the other hand, a settings change in either addon should change the display, which would be harder to pull off if the content warning is done generally in the same item field.